src/sidebar/services/load-annotations.ts

```diff
@@ -175,6 +183,41 @@ export class LoadAnnotationsService {
     this._searchClient.get({ group: groupId, uri: uris });
   }
 
+  /**
+   * If anchoring for an annotation the current set of annotations in the store
+   * fails to complete in a reasonable amount of time, then we assume that the
+   * annotation failed to anchor. If it does later successfully anchor then the
+   * status will be updated.
+   */
+  private _scheduleAnchorStatusCheck() {
+    const ANCHORING_TIMEOUT = 500;
+
+    const anchoringIds = this._store
+      .allAnnotations()
+      .filter(isWaitingToAnchor)
+      .map(ann => ann.id);
+
+    if (anchoringIds.length > 0) {
+      setTimeout(() => {
+        // Find annotations which haven't yet been anchored in the document.
+
+        const annsStillAnchoring = anchoringIds
+          .map(id => (id ? this._store.findAnnotationByID(id) : null))
+          .filter(ann => ann && isWaitingToAnchor(ann));
+
+        // Mark anchoring as timed-out for these annotations.
+        const anchorStatusUpdates = annsStillAnchoring.reduce(
+          (updates, ann) => {
+            updates[ann!.$tag] = 'timeout';
+            return updates;
+          },
+          {} as AnchorStatusUpdates
+        );
+        this._store.updateAnchorStatus(anchorStatusUpdates);
+      }, ANCHORING_TIMEOUT);
```

Comment:
It might be wise to add a `destroy` method to this class to clean up any in-flight activity in tests.